packages/@react-aria/menu/src/useSubmenuTrigger.ts

```diff
@@ -106,6 +106,7 @@ export function UNSTABLE_useSubmenuTrigger<T>(props: AriaSubmenuTriggerProps, st
       case 'Escape':
         e.stopPropagation();
         state.closeAll();
+        ref.current.focus();
```

Comment:
Restoring focus on Escape fails beyond submenu level 2 in the [Submenu Nested Example](https://reactspectrum.blob.core.windows.net/reactspectrum/d4d305826cd9a57226a151185b9be1451237b3ac/storybook/index.html?path=/story/react-aria-components--submenu-nested-example) and [Submenu Many Items Example](https://reactspectrum.blob.core.windows.net/reactspectrum/d4d305826cd9a57226a151185b9be1451237b3ac/storybook/index.html?path=/story/react-aria-components--submenu-many-items-example).

Reply:
@LFDanLu This helps restore focus for the case of one submenu, but doesn't work for the nested case.

Reply:
Would it be easier to manage focus if Escape only closed the top menu, rather than `state.closeAll()`?

Reply:
I've fixed the behavior to restore focus in the nested case. I'll still bring this up with the team on whether we want to close all or one. I think it would be nice if the hooks+RAC supported both cases (i.e. a prop for `shouldCloseAllOnEscape`).

Reply:
After bringing it up with the team, we decided to go with closing all for now. In the future, if the use cases comes up for supporting closing one submenu (via a prop), we can consider it.